pkg/elevate/elevate_test.go

```diff
@@ -0,0 +1,491 @@
+				e: ElevateConfig{
+					GetCurrentUsername:                      GetCurrentUsername,
+					AddImpersonationReasonRoleToCurrentUser: AddImpersonationReasonRoleToCurrentUser,
+					AddElevationReasonToRawKubeconfig:       AddElevationReasonToRawKubeconfig,
+					WriteConfigToTempKubeconfig:             WriteConfigToTempKubeconfig,
+				},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := AddImpersonationReasonRoleToCurrentUser(tt.args.e, tt.args.config); (err != nil) != tt.wantErr {
+				t.Errorf("AddImpersonationReasonRoleToCurrentUser() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestAddElevationReasonToRawKubeconfig(t *testing.T) {
+	type args struct {
+		config          api.Config
+		elevationReason string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "It throws an error if there is no current kubeconfig context",
+			args: args{
+				config:          api.Config{},
+				elevationReason: "Production cluster",
+			},
+			wantErr: true,
+		},
+		{
+			name: "it returns an error if there is no user info in kubeconfig",
+			args: args{
+				config: api.Config{
+					Kind:        "Config",
+					APIVersion:  "v1",
+					Preferences: api.Preferences{},
+					Clusters: map[string]*api.Cluster{
+						"dummy_cluster": {
+							Server: "https://api-backplane.apps.something.com/backplane/cluster/configcluster",
+						},
+					},
+					Contexts: map[string]*api.Context{
+						"default/test123/anonymous": {
+							Cluster:   "dummy_cluster",
+							Namespace: "default",
+						},
+					},
+					CurrentContext: "default/test123/anonymous",
+				},
+				elevationReason: "Production cluster",
+			},
+			wantErr: true,
+		},
+		{
+			name: "it succeeds if the auth info exists for the current context",
+			args: args{
+				config: api.Config{
+					Kind:        "Config",
+					APIVersion:  "v1",
+					Preferences: api.Preferences{},
+					Clusters: map[string]*api.Cluster{
+						"dummy_cluster": {
+							Server: "https://api-backplane.apps.something.com/backplane/cluster/configcluster",
+						},
+					},
+					AuthInfos: map[string]*api.AuthInfo{
+						"anonymous": {
+							LocationOfOrigin: "England",
+						},
+					},
+					Contexts: map[string]*api.Context{
+						"default/test123/anonymous": {
+							Cluster:   "dummy_cluster",
+							Namespace: "default",
+							AuthInfo:  "anonymous",
+						},
+					},
+					CurrentContext: "default/test123/anonymous",
+				},
+				elevationReason: "Production cluster",
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := AddElevationReasonToRawKubeconfig(tt.args.config, tt.args.elevationReason); (err != nil) != tt.wantErr {
+				t.Errorf("AddElevationReasonToRawKubeconfig() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestWriteConfigToTempKubeconfig(t *testing.T) {
```

Comment:
if you can, pls reuse 
https://github.com/openshift/backplane-cli/blob/34e4c2c59fa2af60371ce955ab4e089755bb5a68/pkg/utils/util.go#L162